src/listener/can_open_listener.py

```diff
@@ -105,7 +106,8 @@ def _read_simple_variable(self, sdo_client, sdo_index, node_id):
 
     def _add_nodes(self, nodes):
         for i in range(len(nodes)):
-            self.node_purposes.append(nodes[i]['node_purpose'])
+            self.node_purposes = append(
```

Comment:
is de `self.node_purposes =` nodig? Want je geeft `self.node_purposes` als argument mee

Reply:
Geldt ook voor de andere arrays

Reply:
ja helaas wel, ik had het nog eerder zonder geprobeerd maar je moet het opnieuw setten anders wordt de verandering niet doorgevoerd